### PR TITLE
handle form archive for new ledgers

### PR DESCRIPTION
**Pull request by snopoke**
@benrudolph @czue 

Getting closer. I think the only thing left is hooking up consumption.

:fish:

### CODE REVIEW

**Comment by czue** [comment]
build fail looks real:
```

======================================================================
ERROR: test_delete_ledger_values_raise_error_case (corehq.form_processor.tests.test_ledger_dbaccessor.LedgerDBAccessorTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/mnt/corehq/form_processor/tests/test_ledger_dbaccessor.py", line 33, in tearDown
    FormProcessorTestUtils.delete_all_ledgers(DOMAIN)
  File "/mnt/corehq/form_processor/tests/utils.py", line 49, in delete_all_ledgers
    for case_id in CaseAccessorSQL.get_case_ids_in_domain(domain):
  File "/mnt/corehq/form_processor/backends/sql/dbaccessors.py", line 491, in get_case_ids_in_domain
    cursor.execute('SELECT case_id FROM get_case_ids_in_domain(%s, %s)', [domain, type_])
  File "/vendor/lib/python2.7/site-packages/django/db/backends/utils.py", line 65, in execute
    return self.cursor.execute(sql, params)
  File "/vendor/lib/python2.7/site-packages/django/db/utils.py", line 94, in __exit__
    six.reraise(dj_exc_type, dj_exc_value, traceback)
  File "/vendor/lib/python2.7/site-packages/django/db/backends/utils.py", line 65, in execute
    return self.cursor.execute(sql, params)
InternalError: current transaction is aborted, commands ignored until end of transaction block
```

File: corehq/form_processor/utils/sql.py [hunk 1]
```diff
@@ -4,7 +4,9 @@
 """
 import json
 from collections import namedtuple
+from functools import wraps
 
+from django import db
```
Comment (dimagimon) [inline corehq/form_processor/utils/sql.py]: F401 'db' imported but unused

File: corehq/form_processor/utils/sql.py [hunk 2]
```diff
@@ -4,7 +4,9 @@
 """
 import json
 from collections import namedtuple
+from functools import wraps
```
Comment (dimagimon) [inline corehq/form_processor/utils/sql.py]: F401 'wraps' imported but unused

**Comment by snopoke** [comment]
Damn that was painful to fix!

**Comment by snopoke** [comment]
:white_check_mark:

**Comment by snopoke** [comment]
bump

**Comment by czue** [comment]
this part of the code makes me :dizzy_face: sometimes

**Pull request merged**